src/test/java/com/alphawallet/attestation/PublicAttestationTest.java

```diff
@@ -30,8 +32,30 @@ public static void setupKeys() throws Exception {
     @BeforeEach
     public void makePublicAttestation()
     {
-        Attestation att = HelperTest.makePublicIdAttestation(subjectKeys.getPublic(), "TW", "@kingmidas");
-        attestation = new SignedAttestation(att, attestorKeys);
+        Attestation att2 = HelperTest.makePublicIdAttestation(subjectKeys.getPublic(), "TW", "@kingmidas");
+        attestation = new SignedAttestation(att2, attestorKeys);
+    }
+
+    @Test
+    public void testNFTAttestation() throws Exception
+    {
+        ERC721Token myNFT = new ERC721Token("0xa567f5A165545Fa2639bBdA79991F105EADF8522", "25");
+        NFTAttestation nftAtt = new NFTAttestation(attestation, myNFT);
+        //construct SignedNFTAttestation using subject key
+        nftAttestation = new SignedNFTAttestation(nftAtt, subjectKeys);
+
+        //Extract the Ethereum signature
+        byte[] sig = nftAttestation.getSignature();
+
+        //generate NFTAttestation from the NFTAttestation bytes
+        NFTAttestation nftAttestation2 = new NFTAttestation(nftAtt.getDerEncoding(), attestorKeys.getPublic());
+
+        //check recovered signed attestation within the wrapping
+        assertTrue(nftAttestation2.verify());
+
+        //Generate SignedNFTAttestation using the reconstructed NFTAttestation and the extracted Ethereum signature
+        SignedNFTAttestation signedNFTAttestation2 = new SignedNFTAttestation(nftAttestation2, subjectKeys.getPublic(), sig);
+        assertTrue(signedNFTAttestation2.checkValidity());
```

Comment:
Should probably also have a call `assertTrue(signedNFTAttestation2.verify());`

Reply:
... it does have. Did you mean the original 'nftAttestation' SignedNFTAttestation? I can add that. Both are redundant however, because 'verify()' is called at the end by the constructor.

Reply:
I mean on SignedNFTAttestation. 
It is true that it is redundant since it is currently called during construction. However, tbh I am a bit unhappy abut my initial design choice of always doing so. As I don't think it should be expected by the constructor, and it makes negative testing much harder. Anyway, I am thinking that we might eventually want to remove this constructor check, so calling it here is more about future-proofing and making explicit what is expected of the object.